core/src/main/java/org/elasticsearch/search/internal/ShardSearchLocalRequest.java

```diff
@@ -85,17 +87,19 @@ public ShardSearchLocalRequest(ShardId shardId, String[] types, long nowInMillis
         this.nowInMillis = nowInMillis;
         this.aliasFilter = aliasFilter;
         this.shardId = shardId;
+        indexBoost = 1.0f;
```

Comment:
For example, if we don't set it here, boost is 0. Then, validate query with `explain=true` returns parsed queries with boost 0. (Actually, several tests fails if it's not set here)
So, I leave it as is for now. If it's not right, let me know. I'll fix.